theano/gpuarray/extra_ops.py

```diff
@@ -456,6 +454,8 @@ def c_support_code_struct(self, node, nodename):
 @register_opt2([CumsumOp], 'fast_compile')
 def use_gpu_cumsumop(op, ctx_name, inputs, outputs):
     if inputs[0].dtype == 'float32':
+        if isinstance(inputs[0].type, GpuArrayType):
+            return
```

Comment:
I don't understand. When called in the new optimizer, `inputs` will be the GPU variables, maintained by the global optimizer, that are not part of the graph yet, and that should be the inputs to the new GPU Op.
If you return right away, the optimization will not be applied.